traffic_ops/traffic_ops_golang/deliveryservice/request/requests.go

```diff
@@ -0,0 +1,368 @@
+//if so, it will return an errorType of DataConflict and the type should be appended to the
+//generic error message returned
+
+// Update ...
+func (req *TODeliveryServiceRequest) Update(db *sqlx.DB, user auth.CurrentUser) (error, tc.ApiErrorType) {
+	XMLID, err := req.getXMLID(db)
+	if err != nil {
+		return err, tc.SystemError
+	}
+
+	active, err := isActiveRequest(db, XMLID)
+	if err != nil {
+		return err, tc.SystemError
+	}
+	if active {
+		return errors.New("An active request exists for delivery service '" + XMLID), tc.DataConflictError
+	}
+
+	tx, err := db.Beginx()
+	defer func() {
+		if tx == nil {
+			return
+		}
+		if err == nil {
+			err = tx.Commit()
+		}
+		if err != nil {
+			tx.Rollback()
+		}
+	}()
+
+	if err != nil {
+		log.Error.Println("could not begin transaction: ", err.Error())
+		return err, tc.SystemError
+	}
+	resultRows, err := tx.NamedQuery(updateRequestQuery(), req)
+	if err != nil {
+		if err, ok := err.(*pq.Error); ok {
+			err, eType := dbhelpers.ParsePQUniqueConstraintError(err)
+			if eType == tc.DataConflictError {
+				return errors.New("a request with " + err.Error()), eType
+			}
+			return err, eType
+		}
+		log.Errorf("received error from update execution: %s", err.Error())
+		return err, tc.SystemError
+	}
+	defer resultRows.Close()
+
+	var lastUpdated tc.Time
+	var rowsAffected int
+	for resultRows.Next() {
+		rowsAffected++
+		if err = resultRows.Scan(&lastUpdated); err != nil {
+			log.Error.Println("could not scan lastUpdated from insert: ", err.Error())
+			return err, tc.SystemError
+		}
+	}
+	log.Debugln("lastUpdated: ", lastUpdated)
+	req.LastUpdated = lastUpdated
+	if rowsAffected < 1 {
+		err = errors.New("no request found with this id")
+		return err, tc.DataMissingError
+	}
+	if rowsAffected > 1 {
+		err = fmt.Errorf("this update affected too many rows: %d", rowsAffected)
+		return err, tc.SystemError
+	}
+
+	return nil, tc.NoError
+}
+
+//The TODeliveryServiceRequest implementation of the Inserter interface
+//all implementations of Inserter should use transactions and return the proper errorType
+//ParsePQUniqueConstraintError is used to determine if a request with conflicting values exists
+//if so, it will return an errorType of DataConflict and the type should be appended to the
+//generic error message returned
+//The insert sql returns the id and lastUpdated values of the newly inserted request and have
+//to be added to the struct
+
+// Insert ...
+func (req *TODeliveryServiceRequest) Insert(db *sqlx.DB, user auth.CurrentUser) (error, tc.ApiErrorType) {
+	XMLID, err := req.getXMLID(db)
+	if err != nil {
+		return err, tc.SystemError
+	}
+
+	active, err := isActiveRequest(db, XMLID)
+	if err != nil {
+		return err, tc.SystemError
+	}
+	if active {
+		return errors.New("An active request exists for delivery service '" + XMLID), tc.DataConflictError
+	}
+
+	tx, err := db.Beginx()
+	defer func() {
+		if tx == nil {
+			return
+		}
+		if err == nil {
+			err = tx.Commit()
+		}
```

Comment:
Same issue here wrt to returning nil. The Insert function will report no error, even with Commit errors out.